pandas/core/generic.py

```diff
@@ -9884,6 +9887,9 @@ def _where(
             if cond.shape != self.shape:
                 raise ValueError("Array conditional must be same shape as self")
             cond = self._constructor(cond, **self._construct_axes_dict(), copy=False)
+            # cond may contain NA, see GH #52955
+            # let NA propagate in where and mask operations
+            cond = cond.fillna(True)
 
```

Comment:
Do your changes maker the the lines below unnecessary?

Reply:
I think it didn't. For instance, there is a test
https://github.com/pandas-dev/pandas/blob/016c8686e594d583a2135be90fada554d585db5e/pandas/tests/frame/indexing/test_mask.py#L97

In that test, the line
https://github.com/pandas-dev/pandas/blob/016c8686e594d583a2135be90fada554d585db5e/pandas/core/generic.py#L9879
would make `cond` become
```python
>>> type(cond)
<class 'pandas.core.frame.DataFrame'>
>>> cond
       A
id1  NaN
id2  NaN
id3  NaN
id4  NaN
```
Then it will still go through the `fillna` lines below.

Reply:
Ok thanks for looking into it. Then those lines can be moved to be right after `cond = cond.align(self, join="right", copy=False)[0]` in if statement? (I would ideally like we not to do `fillna` twice when not necessary).